core/src/main/java/com/yahoo/imapnio/async/request/UidFetchCommand.java

```diff
@@ -26,7 +40,19 @@ public UidFetchCommand(@Nonnull final MessageNumberSet[] msgsets, @Nonnull final
      * @param macro the macro, for example, ALL
      */
     public UidFetchCommand(@Nonnull final MessageNumberSet[] msgsets, @Nonnull final FetchMacro macro) {
-        super(true, msgsets, macro);
+        this(msgsets, macro, null);
+    }
+
+    /**
+     * Initializes a {@link UidFetchCommand} with the {@link MessageNumberSet} array.
+     *
+     * @param msgsets the set of message set
+     * @param macro the macro, for example, ALL
+     * @param partialExtUidFetchInfo partial extension uid fetch info
+     */
+    public UidFetchCommand(@Nonnull final MessageNumberSet[] msgsets, @Nonnull final FetchMacro macro,
```

Comment:
not critical but a pattern. Let's not let user to pass in @Nullable partialExtUidFetchInfo, a nullable value makes parameter list looks confusing (required or not required? such). Change this to @Nonnull instead, and let line 21 to call super(...)